debian/upstart

```diff
@@ -0,0 +1,33 @@
+description "drserv is an authenticated HTTP API for publishing .deb package to a repo"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+# never stop trying to respawn
+respawn
+respawn limit unlimited
+
+# Drop privileges
+setuid drserv
+setgid nogroup
+
+env DEFAULTFILE=/etc/default/drserv
+
+pre-start script
+  [ -f $DEFAULTFILE ] && . $DEFAULTFILE
+
+  # don't start unless explicitly enabled in config
+  if [ -z "$ENABLED" ] ; then
```

Comment:
Be careful using ```[[```, upstart uses /bin/sh for shells which does not guarantee the existence of the new test expressions.

```[[ -n "$ENABLED" ]] && $ENABLED``` works with `true` because of it being a command, however it would also work with ```ENABLED="ls"``` which might not be desirable. A command executing with non-zero will suffice for the expression to be true.
The canonical, and portable /bin/sh test would be ```[ "x$ENABLED" = "xtrue" ]```.

```[``` should be a built-in just the same as ```[[``` in all modern shells